File: frontend/src/game/objects/item.ts
import { ANIMS } from "../sprite_animations/animations";
import { MapIndex } from "../types";
import { BaseObj } from "./base";
import type { ItemType } from "../event/events"
export type { ItemType } from "../event/events"
export class ItemObj extends BaseObj {
    constructor(
        scene: Phaser.Scene,
        index: MapIndex,
        itemType: ItemType
    ) {
        super(scene, index, "items", 0, "item");
        this.sprite.anims.play(ANIMS.item[itemType])
    }
}

File: backend2/game/objects/item.go
package objects

import (
	"bomberman/backend2/types"
)

type ItemObj struct {
	Base     BaseObj
	ItemType types.ItemType
}

func NewItemObj(index *types.MapIndex, itemType types.ItemType) *ItemObj {
	return &ItemObj{
		Base:     NewBaseObj(index, types.ObjItem),
		ItemType: itemType,
	}
}
